dusty/systems/virtualbox/__init__.py

```diff
@@ -199,29 +200,55 @@ def get_docker_vm_ip():
             spec = line.split('=')[1].strip('"')
             name, protocol, host, host_port, target, target_port = spec.split(',')
             if name == 'ssh' and protocol == 'tcp' and target_port == '22':
-                ssh_port = host_port
-                break
-
-    if ssh_port:
-        # Now we can SSH to the VM on localhost:ssh_port. We do this to
-        # get inside the VM and determine the VM's local address on
-        # the virtualized network. This IP is what we're ultimately
-        # trying to return from this function.
-        ip_addr = check_output_demoted(['ssh', '-o', 'StrictHostKeyChecking=no',
-                                        '-o', 'UserKnownHostsFile=/dev/null',
-                                        '-i', _vm_key_path(), '-p', ssh_port,
-                                        'docker@127.0.0.1', 'ip addr show dev eth1'])
-        for line in ip_addr.splitlines():
-            line = line.strip()
-            if line.startswith('inet') and not line.startswith('inet6'):
-                ip = line.split(' ')[1].split('/')[0]
-                return ip
-
-    # If all our crazy hacks up there don't work, we can still get the IP
-    # from Machine directly. The reason we don't do this normally is
-    # because it's really slow, taking around 600ms on my fast machine.
-    logging.warning('Could not get IP the fast way, falling back to Docker Machine')
-    return check_output_demoted(['docker-machine', 'ip', constants.VM_MACHINE_NAME]).rstrip()
+                return host_port
+    raise ValueError('Could not determine localhost port for SSH forwarding')
+
+def _get_host_only_mac_address():
+    # Get the number of the host-only adapter
+    vm_config = _get_vm_config()
+    for line in vm_config:
+        if line.startswith('hostonlyadapter'):
+            adapter_number = int(line[15:16])
```

Comment:
I'll go liberally sprinkle comments around once I can get integration to stop flaking out on me